plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -119,12 +114,23 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
         }
     }
 
-    override fun embedFunction(symbol: FirFunctionSymbol<*>): FunctionEmbedding =
-        methods.getOrPut(symbol.embedName(this)) {
-            val signature = embedFullSignature(symbol)
-            val callable = processCallable(symbol, signature)
-            UserFunctionEmbedding(callable)
+    override fun embedFunction(symbol: FirFunctionSymbol<*>): FunctionEmbedding {
```

Comment:
This code here seems to be the reason to have `PartiallySpecialKotlinFunction`.  Let us imagine we decided to only use `SpecialKotlinFunction`, that would sometimes request an embedding.  What piping are we missing for that?  I assume `ProgramConversionContext` would have to be passed somewhere where it currently isn't.

Reply:
I guess you mean that in theory we could only have "partial" functions without "full" ones.
I don't think `ProgramConversionContext` would be needed anywhere else.
But it would create unnecessary `UserFunctionEmbedding`s in simple cases like addition of `Int`s

Reply:
Maybe we should inherit `FullySpecialFunction` from `PartiallySpecialFunction` (and rename them probably)
providing always non-nullable version of `tryInsertCall`.

Reply:
Let's discuss this on the call today.